<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Bisque Example Usage</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Bisque Example Usage</h1>



<p>This vignette provides a basic example of using Bisque to decompose bulk expression. Bisque offers two modes of operation: Reference-based and Marker-based decomposition. We will provide brief examples of both.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Biobase)
<span class="kw">library</span>(BisqueRNA)</code></pre></div>
<div id="input-format" class="section level2">
<h2>Input Format</h2>
<p>Bisque requires expression data in the ExpressionSet format from the Biobase package.</p>
<p>Bulk RNA-seq data can be converted from a matrix (columns are samples, rows are genes) to an ExpressionSet as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bulk.eset &lt;-<span class="st"> </span>Biobase<span class="op">::</span><span class="kw">ExpressionSet</span>(<span class="dt">assayData =</span> bulk.matrix)</code></pre></div>
<p>Single-cell data requires additional information in the ExpressionSet, specificially cell type labels and individual labels. Individual labels indicate which individual each cell originated from. To add this information, Biobase requires it to be stored in a data frame format. Assuming we have character vectors of cell type labels (<code>cell.type.labels</code>) and individual labels (<code>individual.labels</code>), we can convert scRNA-seq data (with counts also in matrix format) as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sample.ids &lt;-<span class="st"> </span><span class="kw">colnames</span>(sc.counts.matrix)
<span class="co"># individual.ids and cell.types should be in the same order as in sample.ids</span>
sc.pheno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">check.names=</span>F, <span class="dt">check.rows=</span>F,
                       <span class="dt">stringsAsFactors=</span>F,
                       <span class="dt">row.names=</span>sample.ids,
                       <span class="dt">SubjectName=</span>individual.labels,
                       <span class="dt">cellType=</span>cell.type.labels)
sc.meta &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">labelDescription=</span><span class="kw">c</span>(<span class="st">&quot;SubjectName&quot;</span>,
                                         <span class="st">&quot;cellType&quot;</span>),
                      <span class="dt">row.names=</span><span class="kw">c</span>(<span class="st">&quot;SubjectName&quot;</span>,
                                  <span class="st">&quot;cellType&quot;</span>))
sc.pdata &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>,
                <span class="dt">data=</span>sc.pheno,
                <span class="dt">varMetadata=</span>sc.meta)
sc.eset &lt;-<span class="st"> </span>Biobase<span class="op">::</span><span class="kw">ExpressionSet</span>(<span class="dt">assayData=</span>sc.counts.matrix,
                                  <span class="dt">phenoData=</span>sc.pdata)</code></pre></div>
<p>If your single-cell data (from 10x platform) is in a Seurat object with cell type assignments, Bisque includes a function that will automatically convert this object to an ExpressionSet:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sc.eset &lt;-<span class="st"> </span>BisqueRNA<span class="op">::</span><span class="kw">SeuratToExpressionSet</span>(seurat.obj, <span class="dt">delimiter=</span><span class="st">&quot;-&quot;</span>, <span class="dt">position=</span><span class="dv">2</span>, <span class="dt">version=</span><span class="st">&quot;v3&quot;</span>)</code></pre></div>
<p>The delimiter and position arguments describe the barcode format of 10x single-cell data. For example, barcodes of “ATCGATCG-1” and “ATGCAAGT-2” have the individual ID in position 2 after splitting by the delimiter ‘-’.</p>
<p>Here is an example of input single-cell and bulk data for 2 individuals with 10 cells sequenced each:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sampleNames</span>(sc.eset)</code></pre></div>
<pre><code>##  [1] &quot;AAAA-1&quot; &quot;TAAA-1&quot; &quot;CAAA-1&quot; &quot;GAAA-1&quot; &quot;ATAA-1&quot; &quot;TTAA-1&quot; &quot;CTAA-1&quot;
##  [8] &quot;GTAA-1&quot; &quot;ACAA-1&quot; &quot;TCAA-1&quot; &quot;AAAA-2&quot; &quot;TAAA-2&quot; &quot;CAAA-2&quot; &quot;GAAA-2&quot;
## [15] &quot;ATAA-2&quot; &quot;TTAA-2&quot; &quot;CTAA-2&quot; &quot;GTAA-2&quot; &quot;ACAA-2&quot; &quot;TCAA-2&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sc.eset<span class="op">$</span>SubjectName</code></pre></div>
<pre><code>##  [1] &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot;
## [18] &quot;2&quot; &quot;2&quot; &quot;2&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sc.eset<span class="op">$</span>cellType</code></pre></div>
<pre><code>##  [1] &quot;Neurons&quot;          &quot;Neurons&quot;          &quot;Neurons&quot;         
##  [4] &quot;Neurons&quot;          &quot;Neurons&quot;          &quot;Neurons&quot;         
##  [7] &quot;Neurons&quot;          &quot;Astrocytes&quot;       &quot;Astrocytes&quot;      
## [10] &quot;Astrocytes&quot;       &quot;Neurons&quot;          &quot;Neurons&quot;         
## [13] &quot;Neurons&quot;          &quot;Neurons&quot;          &quot;Astrocytes&quot;      
## [16] &quot;Astrocytes&quot;       &quot;Astrocytes&quot;       &quot;Astrocytes&quot;      
## [19] &quot;Oligodendrocytes&quot; &quot;Microglia&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sampleNames</span>(bulk.eset)</code></pre></div>
<pre><code>## [1] &quot;1&quot; &quot;2&quot;</code></pre>
<p>Note that if you have samples with both single-cell and bulk RNA-seq data, their IDs should be found in both <code>sc.eset$SubjectName</code> and <code>sampleNames(bulk.eset)</code> .</p>
</div>
<div id="reference-based-decomposition" class="section level2">
<h2>Reference-based decomposition</h2>
<p>We will use data simulated under a simple model (code for SimulateData() can be found in R/simulation.R). We simulate single-cell and bulk RNA-seq counts for 10 individuals. We remove 5 individuals from the single-cell data. We will estimate the cell composition for these 5 individuals.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.types &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Neurons&quot;</span>, <span class="st">&quot;Astrocytes&quot;</span>, <span class="st">&quot;Oligodendrocytes&quot;</span>, <span class="st">&quot;Microglia&quot;</span>, <span class="st">&quot;Endothelial Cells&quot;</span>)
avg.props &lt;-<span class="st"> </span><span class="kw">c</span>(.<span class="dv">5</span>, .<span class="dv">2</span>, .<span class="dv">2</span>, .<span class="dv">07</span>, .<span class="dv">03</span>)
sim.data &lt;-<span class="st"> </span><span class="kw">SimulateData</span>(<span class="dt">n.ind=</span><span class="dv">10</span>, <span class="dt">n.genes=</span><span class="dv">100</span>, <span class="dt">n.cells=</span><span class="dv">500</span>, <span class="dt">cell.types=</span>cell.types, <span class="dt">avg.props=</span>avg.props)
sc.eset &lt;-<span class="st"> </span>sim.data<span class="op">$</span>sc.eset[,sim.data<span class="op">$</span>sc.eset<span class="op">$</span>SubjectName <span class="op">%in%</span><span class="st"> </span><span class="kw">as.character</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">10</span>)]
bulk.eset &lt;-<span class="st"> </span>sim.data<span class="op">$</span>bulk.eset
true.props &lt;-<span class="st"> </span>sim.data<span class="op">$</span>props
markers &lt;-<span class="st"> </span>sim.data<span class="op">$</span>markers</code></pre></div>
<p>By default, Bisque uses all genes for decomposition. However, you may supply a list of genes (such as marker genes) to be used with the <code>markers</code> parameter. Also, since we have samples with both bulk and single-cell RNA-seq data, we set the <code>use.overlap</code> parameter to <code>TRUE</code>. If there are no overlapping samples, you can set this parameter to <code>FALSE</code> (we expect performance to be better if overlapping samples are available).</p>
<p>Here’s how to call the reference-based decomposition method:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span>BisqueRNA<span class="op">::</span><span class="kw">ReferenceBasedDecomposition</span>(bulk.eset, sc.eset, <span class="dt">markers=</span><span class="ot">NULL</span>, <span class="dt">use.overlap=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Decomposing into 5 cell types.</code></pre>
<pre><code>## Found 5 samples with bulk and single-cell expression.</code></pre>
<pre><code>## Remaining 5 bulk samples will be decomposed.</code></pre>
<pre><code>## Using 100 genes in both bulk and single-cell expression.</code></pre>
<pre><code>## Converting single-cell counts to CPM and filtering zero variance genes.</code></pre>
<pre><code>## Filtered 6 zero variance genes.</code></pre>
<pre><code>## Converting bulk counts to CPM and filtering unexpressed genes.</code></pre>
<pre><code>## Filtered 6 unexpressed genes.</code></pre>
<pre><code>## Generating single-cell based reference from 2500 cells.</code></pre>
<pre><code>## Learning bulk transformation from overlapping samples.</code></pre>
<pre><code>## Applying transformation to bulk samples and decomposing.</code></pre>
<p>A list is returned with decomposition estimates in slot <code>bulk.props</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ref.based.estimates &lt;-<span class="st"> </span>res<span class="op">$</span>bulk.props
knitr<span class="op">::</span><span class="kw">kable</span>(ref.based.estimates, <span class="dt">digits=</span><span class="dv">2</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Astrocytes</td>
<td align="right">0.19</td>
<td align="right">0.18</td>
<td align="right">0.19</td>
<td align="right">0.20</td>
<td align="right">0.18</td>
</tr>
<tr class="even">
<td>Endothelial Cells</td>
<td align="right">0.03</td>
<td align="right">0.04</td>
<td align="right">0.03</td>
<td align="right">0.03</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td>Microglia</td>
<td align="right">0.09</td>
<td align="right">0.07</td>
<td align="right">0.06</td>
<td align="right">0.08</td>
<td align="right">0.07</td>
</tr>
<tr class="even">
<td>Neurons</td>
<td align="right">0.50</td>
<td align="right">0.54</td>
<td align="right">0.52</td>
<td align="right">0.50</td>
<td align="right">0.51</td>
</tr>
<tr class="odd">
<td>Oligodendrocytes</td>
<td align="right">0.20</td>
<td align="right">0.17</td>
<td align="right">0.20</td>
<td align="right">0.20</td>
<td align="right">0.20</td>
</tr>
</tbody>
</table>
<p>Just to make sure this worked, we can correlate all the estimates with the true proportions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">r &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">as.vector</span>(ref.based.estimates), 
         <span class="kw">as.vector</span>(true.props[<span class="kw">row.names</span>(ref.based.estimates),<span class="kw">colnames</span>(ref.based.estimates)]))
knitr<span class="op">::</span><span class="kw">knit_print</span>(<span class="kw">sprintf</span>(<span class="st">&quot;R: %f&quot;</span>, r))</code></pre></div>
<pre><code>## [1] &quot;R: 0.998478&quot;</code></pre>
</div>
<div id="marker-based-decomposition" class="section level2">
<h2>Marker-based decomposition</h2>
<p>BisqueMarker can provide estimates of relative cell type abundances using only known marker genes when a reference profile is not available. Marker genes are stored in a data frame with columns that specify gene, cluster that the gene is a marker for, and an optional column for weights (typically fold-change). Here’s what this data frame might look like:</p>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="left">cluster</th>
<th align="right">avg_logFC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Gene 1</td>
<td align="left">Neurons</td>
<td align="right">0.82</td>
</tr>
<tr class="even">
<td align="left">Gene 2</td>
<td align="left">Neurons</td>
<td align="right">0.59</td>
</tr>
<tr class="odd">
<td align="left">Gene 3</td>
<td align="left">Astrocytes</td>
<td align="right">0.68</td>
</tr>
<tr class="even">
<td align="left">Gene 4</td>
<td align="left">Oligodendrocytes</td>
<td align="right">0.66</td>
</tr>
<tr class="odd">
<td align="left">Gene 5</td>
<td align="left">Microglia</td>
<td align="right">0.71</td>
</tr>
<tr class="even">
<td align="left">Gene 6</td>
<td align="left">Endothelial Cells</td>
<td align="right">0.62</td>
</tr>
</tbody>
</table>
<p>Here’s how to call the marker-based decomposition method:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span>BisqueRNA<span class="op">::</span><span class="kw">MarkerBasedDecomposition</span>(bulk.eset, markers, <span class="dt">weighted=</span>F)</code></pre></div>
<pre><code>## Getting unique markers</code></pre>
<pre><code>## Estimating proportions for 5 cell types in 10 samples</code></pre>
<pre><code>## Filtered 6 zero variance genes.</code></pre>
<pre><code>## Using 6 genes for cell type Astrocytes;</code></pre>
<pre><code>## 100% of 6 marker genes correlate positively with PC1 for cell type Astrocytes</code></pre>
<pre><code>## Using 6 genes for cell type Endothelial Cells;</code></pre>
<pre><code>## 100% of 6 marker genes correlate positively with PC1 for cell type Endothelial Cells</code></pre>
<pre><code>## Using 5 genes for cell type Microglia;</code></pre>
<pre><code>## 100% of 5 marker genes correlate positively with PC1 for cell type Microglia</code></pre>
<pre><code>## Using 15 genes for cell type Neurons;</code></pre>
<pre><code>## 93% of 15 marker genes correlate positively with PC1 for cell type Neurons</code></pre>
<pre><code>## Using 10 genes for cell type Oligodendrocytes;</code></pre>
<pre><code>## 100% of 10 marker genes correlate positively with PC1 for cell type Oligodendrocytes</code></pre>
<pre><code>## Finished estimating cell type proportions using PCA</code></pre>
<p>A list is returned with decomposition estimates in slot <code>bulk.props</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">marker.based.estimates &lt;-<span class="st"> </span>res<span class="op">$</span>bulk.props
knitr<span class="op">::</span><span class="kw">kable</span>(marker.based.estimates, <span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
<th align="right">6</th>
<th align="right">7</th>
<th align="right">8</th>
<th align="right">9</th>
<th align="right">10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Astrocytes</td>
<td align="right">0.35</td>
<td align="right">-0.93</td>
<td align="right">0.84</td>
<td align="right">3.46</td>
<td align="right">-2.17</td>
<td align="right">-0.62</td>
<td align="right">-0.13</td>
<td align="right">-0.93</td>
<td align="right">2.93</td>
<td align="right">-2.79</td>
</tr>
<tr class="even">
<td>Endothelial Cells</td>
<td align="right">1.78</td>
<td align="right">-0.46</td>
<td align="right">1.52</td>
<td align="right">-0.29</td>
<td align="right">2.55</td>
<td align="right">-0.97</td>
<td align="right">-2.57</td>
<td align="right">0.05</td>
<td align="right">-1.71</td>
<td align="right">0.10</td>
</tr>
<tr class="odd">
<td>Microglia</td>
<td align="right">1.73</td>
<td align="right">-0.18</td>
<td align="right">-2.01</td>
<td align="right">0.70</td>
<td align="right">0.07</td>
<td align="right">-1.65</td>
<td align="right">2.91</td>
<td align="right">-0.65</td>
<td align="right">1.98</td>
<td align="right">-2.90</td>
</tr>
<tr class="even">
<td>Neurons</td>
<td align="right">-2.99</td>
<td align="right">4.48</td>
<td align="right">1.00</td>
<td align="right">-2.56</td>
<td align="right">-1.55</td>
<td align="right">1.32</td>
<td align="right">1.43</td>
<td align="right">-2.68</td>
<td align="right">0.35</td>
<td align="right">1.20</td>
</tr>
<tr class="odd">
<td>Oligodendrocytes</td>
<td align="right">1.71</td>
<td align="right">-3.74</td>
<td align="right">-0.18</td>
<td align="right">-0.53</td>
<td align="right">1.40</td>
<td align="right">0.98</td>
<td align="right">-2.64</td>
<td align="right">3.21</td>
<td align="right">-2.67</td>
<td align="right">2.46</td>
</tr>
</tbody>
</table>
<p>Note that these estimates are relative within each cell type, so you cannot immediately compare abundance estimates between cell types.</p>
<p>Just to make sure this worked, we can correlate these estimates with the scaled true proportions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scaled.true.props &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">scale</span>(<span class="kw">t</span>(true.props)))[<span class="kw">rownames</span>(marker.based.estimates),]
r &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">as.vector</span>(marker.based.estimates),
         <span class="kw">as.vector</span>(scaled.true.props))
knitr<span class="op">::</span><span class="kw">knit_print</span>(<span class="kw">sprintf</span>(<span class="st">&quot;R: %f&quot;</span>, r))</code></pre></div>
<pre><code>## [1] &quot;R: 0.924982&quot;</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
